python/cudf_polars/cudf_polars/dsl/utils/windows.py

```diff
@@ -0,0 +1,197 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Utilities for rolling window aggregations."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import pyarrow as pa
+
+import polars as pl
+
+import pylibcudf as plc
+
+if TYPE_CHECKING:
+    from cudf_polars.typing import ClosedInterval, Duration
+
+
+__all__ = [
+    "duration_to_int",
+    "duration_to_scalar",
+    "offsets_to_windows",
+    "range_window_bounds",
+]
+
+
+def duration_to_int(
+    dtype: plc.DataType,
+    months: int,
+    weeks: int,
+    days: int,
+    nanoseconds: int,
+    parsed_int: bool,  # noqa: FBT001
+    negative: bool,  # noqa: FBT001
+) -> int:
```

Comment:
I would like a `Duration` object in libcudf so I can say `Add 1 week to this date`, but that doesn't exist so we need to convert to single `duration`s.